src/integrationTest/java/uk/gov/hmcts/darts/common/service/AutomatedTaskServiceTest.java

```diff
@@ -0,0 +1,136 @@
+package uk.gov.hmcts.darts.common.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.config.CronTask;
+import org.springframework.scheduling.config.FixedDelayTask;
+import org.springframework.scheduling.config.FixedRateTask;
+import org.springframework.scheduling.config.ScheduledTask;
+import org.springframework.scheduling.config.ScheduledTaskHolder;
+import org.springframework.scheduling.config.Task;
+import org.springframework.scheduling.config.TriggerTask;
+import uk.gov.hmcts.darts.common.entity.AutomatedTaskEntity;
+import uk.gov.hmcts.darts.common.exception.DartsApiException;
+import uk.gov.hmcts.darts.common.task.AutomatedTask;
+import uk.gov.hmcts.darts.common.task.ProcessDailyListAutomatedTask;
+import uk.gov.hmcts.darts.testutils.IntegrationBase;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Slf4j
+class AutomatedTaskServiceTest extends IntegrationBase {
+    @Autowired
+    private AutomatedTaskService automatedTaskService;
+
+    @Autowired
+    private ScheduledTaskHolder scheduledTaskHolder;
+
+    @Test
+    void givenConfiguredTasksCancelTaskAndAddTaskBackWithRandomCronExpression() throws InterruptedException {
```

Comment:
I think this test is trying to do too much, so it's quite hard to read. Can be it split up into multiple tests, each testing (and asserting) a specific part of the functionality? E.g:

- A test to demonstrate task cancellation works
- A test to demonstrate updating the cron expression works
- A test to demonstrate reloading works
..etc